tests/ui/vec.rs

```diff
@@ -75,4 +81,44 @@ fn main() {
     for a in vec![1; 201] {
         println!("{:?}", a);
     }
+
+    // Do lint
+    let mut x = vec![1, 2, 3];
+    x.fill(123);
+    dbg!(x[0]);
```

Comment:
One interesting case I just thought of is that changing a vec to an array can sometimes lead to compile errors when indexing with a constant that is out of bounds, because the compiler checks index expressions for arrays but not vecs:

```rs
let ve = vec![1, 2];
ve[3]; // ok (no compile error)

let arr = [1, 2];
arr[3]; // compile error
```

Not really sure what to do here. Accept it as a "known issue"? I doubt this is something that people run into by accident?

Reply:
"Known issue" is likely fine IMO, this would likely be helpful anyway

Reply:
works for me